src/examples/renaming_and_moving_objects.cpp

```diff
@@ -0,0 +1,75 @@
+#include <H5File.hpp>
+#include <H5Group.hpp>
+#include <H5DataSet.hpp>
+#include <H5DataSpace.hpp>
+#include <H5Attribute.hpp>
+
+using namespace HighFive;
+
+int main(int argc, char *argv[])
+{
+    /* We are going to create group in root directory then add
+     * dataset to this group and attach attribute to the dataset.
+     * Then we are trying to get path to the root, group dataset
+     * and the name of the dataset.
+     * Secondly we will move dataset with attached attribute to
+     * some destination path. Every '/' sign means that a group
+     * will be created there (it is done automatically).
+     * When we have moved the dataset we can check if dataset
+     * object is still valid? We do this by creating second
+     * attribute. There is a tricky part at the end when we
+     * move Group with its dataset */
+
+    // Create a new file using the default property lists.
+    HighFive::File file("names.h5", File::ReadWrite | File::Create | File::Truncate);
```

Comment:
```suggestion
   File file("names.h5", File::ReadWrite | File::Create | File::Truncate);
```
Since you're using the namespace (which personally I hate, but ok, the same is done in other examples.